presto-hive/src/main/java/io/prestosql/plugin/hive/HiveQlToPrestoTranslator.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.PeekingIterator;
+import io.prestosql.spi.PrestoException;
+
+import static com.google.common.collect.Iterators.peekingIterator;
+import static io.prestosql.plugin.hive.HiveErrorCode.HIVE_VIEW_TRANSLATION_ERROR;
+import static java.lang.String.format;
+import static org.apache.hadoop.hive.ql.parse.BaseSemanticAnalyzer.unescapeSQLString;
+
+/**
+ * Translate statements in Hive QL to Presto SQL.
+ *
+ * Only translation of quoted literals is currently included.
+ */
+public final class HiveQlToPrestoTranslator
+{
+    // Translation methods consume data from the iterator
+    private final PeekingIterator<Character> input;
+    private final StringBuilder output = new StringBuilder();
+
+    /**
+     * Translate a HiveQL statement to Presto SQL by fixing quoted identifiers
+     * and string literals. No other translation is performed.
+     *
+     * <p>Backquotes are replaced with double quotes, including SQL-style
+     * escapes (`` becomes ` and " becomes "").
+     *
+     * <p>Single and double quotes are replaced with single quotes, with
+     * minimal processing of escape sequences to ensure that the strings end in
+     * the right place.
+     */
+    public static String translateHiveViewToPresto(String hiveStatement)
+    {
+        HiveQlToPrestoTranslator translator = new HiveQlToPrestoTranslator(hiveStatement);
+        return translator.translateQuotedLiterals();
+    }
+
+    private HiveQlToPrestoTranslator(String hiveQl)
+    {
+        input = peekingIterator(Lists.charactersOf(hiveQl).iterator());
+    }
+
+    private String translateQuotedLiterals()
+    {
+        // Consume input, passing control to other translation methods when
+        // their delimiters are encountered.
+        while (input.hasNext()) {
+            char c = input.next();
+            switch (c) {
+                case '"':
+                case '\'':
+                    translateString(c);
+                    break;
+                case '`':
+                    translateQuotedIdentifier();
+                    break;
+                default:
+                    output.append(c);
+                    break;
+            }
+        }
+
+        return output.toString();
+    }
+
+    private void translateString(char delimiter)
+    {
+        output.append("'");
```

Comment:
Move this to be with the other output append, so that it's easy to see the quotes are matching.